### PR TITLE
Decompile daycare

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

**Comment by DizzyEggg** [comment]
Someone may need to take a look and fix the nonmatchings in Ruby.

File: src/daycare.c [hunk 1]
```diff
@@ -0,0 +1,1316 @@
...
+            break;
+    }
+
+    return species;
+}
+
+static s32 GetSlotToInheritNature(struct DayCare *daycare)
+{
+    u32 species[DAYCARE_MON_COUNT];
+    s32 i, slot = -1, dittoCount;
```
Comment (Diegoisawesome) [inline src/daycare.c]: If this formatting isn't necessary, please change it.

File: src/daycare.c [hunk 2]
```diff
@@ -0,0 +1,1316 @@
...
+        if (GetBoxMonGender(&daycare->mons[i].mon) == MON_FEMALE)
+            slot = i;
+    }
+
+    // search for ditto
+    for (dittoCount = 0, i = 0; i < DAYCARE_MON_COUNT; i++)
+    {
+        species[i] = GetBoxMonData(&daycare->mons[i].mon, MON_DATA_SPECIES);
+        if (species[i] == SPECIES_DITTO)
+            dittoCount++, slot = i;
```
Comment (Diegoisawesome) [inline src/daycare.c]: Here too

File: src/daycare.c [hunk 3]
```diff
@@ -0,0 +1,1316 @@
...
+    }
+
+    return 0;
+}
+
+static u8 GetDaycarePokemonCount(void)
+{
+    u8 ret = CountPokemonInDaycare(&gSaveBlock1Ptr->daycare);
+    if (ret)
+        return ret;
```
Comment (Diegoisawesome) [inline src/daycare.c]: What a useless check, lol

File: include/data/pokemon/egg_moves.h [hunk 4]
```diff
@@ -0,0 +1,1315 @@
+#ifndef POKEEMERALD_DAYCARE_H
+#define POKEEMERALD_DAYCARE_H
+
+#define EGG_MOVES_SPECIES_OFFSET 20000
+#define EGG_MOVES_TERMINATOR 0xFFFF
+#define egg_moves(species, moves...) (SPECIES_##species + EGG_MOVES_SPECIES_OFFSET), moves
+
+#endif // POKEEMERALD_DAYCARE_H
```
[the remark this answers is not in the document]
Comment (DizzyEggg) [reply include/data/pokemon/egg_moves.h]: taken from pokeruby

File: src/daycare.c (hunk 1 above)
Comment (DizzyEggg) [reply src/daycare.c]: This I can change.

File: src/daycare.c (hunk 2 above)
Comment (DizzyEggg) [reply src/daycare.c]: But this? Pls, it allows me to write it in one line without brackets.

Comment (Diegoisawesome) [reply src/daycare.c]: Yes, but writing it on one line isn't the usual style. It also become slightly less clear.

**Pull request merged**